File: backend/src/test/java/com/skewwhiffy/auraltester/fraction/FactorsTest.java
package com.skewwhiffy.auraltester.fraction;

import lombok.val;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

public class FactorsTest {
    @ParameterizedTest
    @CsvSource({
            "11, 12, 132, 1",
            "4, 6, 12, 2",
            "-4, 6, 12, 2",
            "4, -6, 12, 2",
            "-4, -6, 12, 2"
    })
    void when_negativeInput_then_hcfAndLcmWork(int first, int second, int lcm, int hcf) {
        val actualLcm = Factors.lcm(first, second);
        val actualHcf = Factors.hcf(first, second);

        assertThat(actualLcm).isEqualTo(lcm);
        assertThat(actualHcf).isEqualTo(hcf);    }
}

File: backend/src/test/kotlin/FactorsTest.kt
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class FactorsTest {
    @ParameterizedTest
    @CsvSource(
        "11, 12, 132, 1",
        "4, 6, 12, 2",
        "-4, 6, 12, 2",
        "4, -6, 12, 2",
        "-4, -6, 12, 2"
    )
    fun `when negative input then hcf and lcm work`(first: Int, second: Int, lcm: Int, hcf: Int) {
        val actualLcm = Factors.lcm(first, second)
        val actualHcf = Factors.hcf(first, second)

        assertThat(actualLcm).isEqualTo(lcm)
        assertThat(actualHcf).isEqualTo(hcf)
    }
}